src/TestSuite/Constraint/Session/SessionEquals.php

```diff
@@ -65,7 +66,10 @@ public function __construct(?Session $session, string $path)
      */
     public function matches($other): bool
     {
-        return $this->session->read($this->path) === $other;
+        // Server::run calls Session::close at the end of the request.
+        // Which means, that we cannot use Session object here to access the session data.
+        // Call to Session::read will start new session (and will erase the data).
+        return Hash::get($_SESSION, $this->path) === $other;
```

Comment:
Directly using `$_SESSION` here means one will never be able to mock the session class. Though I don't have an alternative suggestion.

Reply:
If you're using a mock, you can't use these methods anyways.

Reply:
I'm not sure what did you mean by "mock the session class", but I can try to offer a solution.

`Session` currently has few `isCLI` if-s, that are true while running tests. It means that it basically uses different implementation for CLI/tests. I can imagine having completely different classes for request/cli/test sessions. And the test one could allow us to read its contents here.

Reply:
I was referring to the scenario where a test has mocked the session. In that case you couldn't really have used these methods before either as the read() or write() method would likely have been mocked.